docusaurus/docs/tutorials/converting-a-frontend-datasource-to-backend.md

```diff
@@ -0,0 +1,370 @@
+title: Converting a Frontend data source to have a Backend Component
+sidebar_position: 10
+description: Learn how to convert a frontend-only data source to a backend data source.
+keywords:
+  - grafana
+  - plugins
+  - plugin
+  - backend
+  - frontend
+  - datasource
+---
+
+# Converting a Frontend data source to have a Backend Component
+
+This guide walks you through the process of converting an existing frontend only data source plugin into a [backend plugin](https://grafana.com/developers/plugin-tools/introduction/backend-plugins).
+
+To convert a frontend data source, we recommend scaffolding a new backend data source plugin using `npx @grafana/create-plugin@latest`, you can then use the following information to extend this foundation to copy over functionality from your original plugin.
+
+## Primary data source components
+
+Before going into specific conversion advice, we will cover the main components of a data source and how these differ between frontend and backend plugins.
+
+### Frontend Datasource Class
+
+Data source plugins will implement a new DataSourcePlugin which will take as a parameter a DataSource class, which for frontend data sources will extend `DataSourceApi`, and for backend data sources will instead extend `DataSourceWithBackend`. The `DatasourceWithBackend` class already implements most of the required methods so migrating to this will highly simplify the plugin frontend code.
+
+Data source plugins require two components - a Query Editor and a Config Editor.
+
+Examples:
+
+- [Frontend data source](https://github.com/grafana/grafana-plugin-examples/blob/main/examples/datasource-http/src/DataSource.ts#L14)
+- [Backend data source](https://github.com/grafana/grafana-plugin-examples/blob/main/examples/datasource-http-backend/src/datasource.ts#L6)
+
+### Query Editor
+
+The query editor allows users to construct a query from a dashboard panel or the Explore view.
+
+This is a frontend component so no changes are _required_. However, by adding a backend component to a data source and thereby being able to make use of `CallResource`, you could potentially enhance a query editor with additional features - such as schema validation.
+
+### Config Editor
+
+The config editor allows users to configure a connection and create a data source instance.
+
+This is a frontend component so no changes are _required_. Config Editors may store sensitive information in `secureJsonData`, and backend plugins can access this information as required - for example as part of handling authentication to the downstream service.
+
+## Introduction to backend plugins
+
+Before we dive into the specifics of converting a frontend data source to a backend data source, it would be better to familiarize yourself with the process of creating a backend data source plugin. Please follow the [backend plugin](https://grafana.com/developers/plugin-tools/introduction/backend-plugins) introduction before proceeding with this tutorial.
```

Comment:
@sympatheticmoose rather than talking about the file / folder structure here I just recommended to learn more about backend plugins. WDYT?

Reply:
That was not the link I wanted :) So what I have done now is to move this paragraph to the introduction and point to the tutorial to build a backend ds plugin. I have also restored the section for the folder structure.

> What's the reason for wanting to drop entirely?

Thought that the info is a bit duplicated if people already knows how to build / the parts of a backend datasource, which is in other pages but I am okay leaving it